### ISSUE TITLE
Timeout when calling SetScan after SetAddAsync in 1.1.572

### ISSUE THREAD

**Issue by rossipedia** [StackExchange/StackExchange.Redis#401]
Sample repro code (Console App):

``` csharp
public class Program
{
    public static void Main()
    {
        try
        {
            Task.Run(Run).Wait();
        }
        catch (Exception e)
        {
            Console.WriteLine(e.ToString());
        }
    }

    static async Task Run()
    {
        using (var redis = await ConnectionMultiplexer.ConnectAsync("localhost:6379"))
        {
            var db = redis.GetDatabase();
            await db.SetAddAsync("Foo", "Bar");
            foreach (var item in db.SetScan("Foo"))
                Console.WriteLine(item);
        }
    }
}
```

Error thrown:

```
System.AggregateException: One or more errors occurred. ---> System.TimeoutException: Timeout performing SSCAN Foo, inst: 1, mgr: ProcessReadQueue, err: never, queue: 2, qu: 0, qs: 2, qc: 0, wr: 0, wq: 0, in: 31, ar: 1, clientName: REM-BROSS-DT, IOCP: (Busy=0,Free=1000,Min=12,Max=1000), WORKER: (Busy=0,Free=2047,Min=12,Max=2047)
   at StackExchange.Redis.ConnectionMultiplexer.ExecuteSyncImpl[T](Message message, ResultProcessor`1 processor, ServerEndPoint server) in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\ConnectionMultiplexer.cs:line 1982
   at StackExchange.Redis.RedisBase.ExecuteSync[T](Message message, ResultProcessor`1 processor, ServerEndPoint server) in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\RedisBase.cs:line 81
   at StackExchange.Redis.RedisBase.CursorEnumerable`1.GetNextPageSync(IScanningCursor obj, Int64 cursor) in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\RedisBase.cs:line 204
   at StackExchange.Redis.RedisBase.CursorEnumerable`1.CursorEnumerator.MoveNext() in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\RedisBase.cs:line 276
   at ScratchConsole.Program.<Run>d__1.MoveNext() in D:\Dev\ScratchConsole\Program.cs:line 27
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at ScratchConsole.Program.Main() in D:\Dev\ScratchConsole\Program.cs:line 13
---> (Inner Exception #0) System.TimeoutException: Timeout performing SSCAN Foo, inst: 1, mgr: ProcessReadQueue, err: never, queue: 2, qu: 0, qs: 2, qc: 0, wr: 0, wq: 0, in: 31, ar: 1, clientName: REM-BROSS-DT, IOCP: (Busy=0,Free=1000,Min=12,Max=1000), WORKER: (Busy=0,Free=2047,Min=12,Max=2047)
   at StackExchange.Redis.ConnectionMultiplexer.ExecuteSyncImpl[T](Message message, ResultProcessor`1 processor, ServerEndPoint server) in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\ConnectionMultiplexer.cs:line 1982
   at StackExchange.Redis.RedisBase.ExecuteSync[T](Message message, ResultProcessor`1 processor, ServerEndPoint server) in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\RedisBase.cs:line 81
   at StackExchange.Redis.RedisBase.CursorEnumerable`1.GetNextPageSync(IScanningCursor obj, Int64 cursor) in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\RedisBase.cs:line 204
   at StackExchange.Redis.RedisBase.CursorEnumerable`1.CursorEnumerator.MoveNext() in C:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\RedisBase.cs:line 276
   at ScratchConsole.Program.<Run>d__1.MoveNext() in D:\Dev\ScratchConsole\Program.cs:line 27<---
```

Also, note the above code works fine on version `1.0.488`

**Comment by rossipedia** [comment]
Verified still throwing a timeout exception on `v1.0.602`

**Comment by rossipedia** [comment]
Changing the code to:

``` c#
await db.SetAddAsync("Foo", "Bar");
await Task.Delay(1);
foreach (var item in db.SetScan("Foo"))
    Console.WriteLine(item);
```

Allows the SetScan to complete correctly.

**Comment by mgravell** [comment]
Note: fails in net46; works fine in net452; initial hunch: the change of task completion mechanism

**Comment by NickCraver** [comment]
Fixed with the task changes in 1.1.605, closing to cleanup!
